src/merkle/store.rs

```diff
@@ -99,6 +99,10 @@ impl MerkleStore {
         I: IntoIterator<Item = Word>,
     {
         let leaves: Vec<_> = leaves.into_iter().collect();
+        if leaves.len() < 2 {
+            return Err(MerkleError::DepthTooSmall(leaves.len() as u8));
+        }
+
         let layers = leaves.len().ilog2();
```

Comment:
nice catch, an alternative is to move this line down by one, the same check is performed by `MerkleTree::new`. would also be nice to have a test for this.